### ISSUE TITLE
SQS config problem

### ISSUE THREAD

**Issue by jameshiscott** [rgraciano/echo-sonos#156]
Hi,

Got so close to getting this to work. The setup works fine if i do a port forward but i have have now tried to add the SQS part in and must be missing something.

Both services (SQS and Sonos http api) are running on the same PI3 both the SQS service and Lambda server are set to run in us-east-1.

When i start the SQS server i dont get any errors and the queues in SQS console (SQL-Procy-Client & SQL-Proxy-Server) so that all looks good. It also has the additional role added.

I changed the Environment variables on my Lambda item to say:

![image](https://cloud.githubusercontent.com/assets/3490711/25090394/cf09bf4a-237a-11e7-84c4-95916f3d3111.png)

*APPID, Guids etc removed from screenshot and logs

When i try to test with the above i get:

```
START RequestId: 95bee9b2-2372-11e7-83f9-5f0049496c4f Version: $LATEST
2017-04-17T13:34:30.381Z	95bee9b2-2372-11e7-83f9-5f0049496c4f	session applicationId: amzn1.ask.skill.XXXXXXXXXXXXXX
2017-04-17T13:34:30.381Z	YYYYYYYYYYY	dispatch intent = PauseAllIntent
2017-04-17T13:34:30.381Z	YYYYYYYYYYY	PauseAllIntent received
2017-04-17T13:34:30.381Z	YYYYYYYYYYY	Calling '/pauseall'
2017-04-17T13:34:30.382Z	YYYYYYYYYYY	Call to 'https://sqs.us-east-1.amazonaws.com/1234567/pauseall' failed
2017-04-17T13:34:30.382Z	YYYYYYYYYYY		{ Error: connect ECONNREFUSED 127.0.0.1:5005
    at Object.exports._errnoException (util.js:1018:11)
    at exports._exceptionWithHostPort (util.js:1041:20)
```
So it looks like it is trying to get to the SQS service. if i go to the URL i simply get:

<UnknownOperationException/>

However you can also see later in the log it trys to go to the HOST name as well.

I tried leaving the HOST name as the old host name (when i had firewall forwarding) but that still gave a timeout with looking like it goes to the SQS event at all.

```
START RequestId: YYYYYYYYYYY Version: $LATEST
2017-04-17T13:38:48.595Z	YYYYYYYYYYY	session applicationId: amzn1.ask.skill.XXXXXXXXXX
2017-04-17T13:38:48.596Z	YYYYYYYYYYY	dispatch intent = PauseAllIntent
2017-04-17T13:38:48.596Z	YYYYYYYYYYY	PauseAllIntent received
2017-04-17T13:38:48.596Z	YYYYYYYYYYY	Calling '/pauseall'
END RequestId: YYYYYYYYYYY
REPORT RequestId: YYYYYYYYYYY	Duration: 30003.08 ms	Billed Duration: 30000 ms 	Memory Size: 128 MB	Max Memory Used: 27 MB	
2017-04-17T13:39:18.596Z YYYYYYYYYYY Task timed out after 30.00 seconds
```

From looking at the logs on the SQS the client is getting some "emptyReceives" but nothing else.

![image](https://cloud.githubusercontent.com/assets/3490711/25090624/33e90334-237c-11e7-8ba8-ccf9bce44399.png)

If i ask Echo directly to do anything "Alexa ask sonos to pause all" i get a full tone from her and that is all. Nothing in the app no "sorry i didnt get that" etc.

Anyone able to help?

**Comment by pheintzelman** [comment]
Did you setup the sqs config file?

**Comment by jameshiscott** [comment]
Yep, and i can see its using the key correctly, it comes up as last used today and used with SQS. 

![image](https://cloud.githubusercontent.com/assets/3490711/25332091/41fea7f6-28dd-11e7-9756-832e577307e6.png)

And making the items in SQS console:
![image](https://cloud.githubusercontent.com/assets/3490711/25332035/19e0fb34-28dd-11e7-83de-7283fa2f269f.png)

**Comment by jameshiscott** [comment]
Its like it is ignoring the USE_SQS = true flag.

**Comment by pheintzelman** [comment]
you could try changing getDefault('USE_SQS', false) in options.js to true

I use sqs but have not set up the host vars. Maybe true in the setting is being passed in as a string.

**Comment by jameshiscott** [comment]
Yep i even tried that and i get as per the above (and below):

I now only have USE_SQS and APPID as environment variables.

```
2017-04-26T07:56:32.823Z	YYYYYY-YYYYYY-YYYYY	Call to 'https://sqs.us-east-1.amazonaws.com/743977751271/pauseall' failed
2017-04-26T07:56:32.823Z	YYYYYY-YYYYYY-YYYYY	{ Error: getaddrinfo ENOTFOUND yourhosthere yourhosthere:5005
    at errnoException (dns.js:28:10)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:76:26)
  code: 'ENOTFOUND',
  errno: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'yourhosthere',
  host: 'yourhosthere',
  port: '5005' }
```

Is the attempt to talk with **https://sqs.us-east-1.amazonaws.com/743977751271/pauseall** correct, it seems to do it even if i put false in the USE_SQS setting.

Should i just scrap it and start again?

**Comment by youdy** [comment]
Did you manage to sort this? Im receiving the same error

**Comment by jameshiscott** [comment]
nope, full reinstall, same problem. Please let me know if you sort it, driving me mad.

**Comment by GregRocket** [comment]
I have SQS working. Environment variables are working for me.
Here is a screenshot of my Environment variables settings:
![firefox_2017-05-01_07-07-42](https://cloud.githubusercontent.com/assets/12052578/25582024/e89490dc-2e3f-11e7-953e-3cfc728ee47d.png)
The HOST, AUTH_USERNAME and AUTH_PASSWORD setting above are not used by SQS. These settings are ignored by SQS.

And here is my directory setup on my PI3.
![putty_2017-05-01_07-10-46a](https://cloud.githubusercontent.com/assets/12052578/25582027/eeb6d77c-2e3f-11e7-8bcd-af5ca8acc44d.png)

When I look at your posts this seems odd to me { Error: connect ECONNREFUSED 127.0.0.1:5005 The IP is what I see that looks wrong but I am not sure that is the problem. Is this really the IP of your PI?

**Comment by youdy** [comment]
The local loopback is 127.0.0.1 however the ethernet is 192.168.0.36

**Comment by jameshiscott** [comment]
@youdy did you get any closer?

**Comment by youdy** [comment]
@jameshiscott unfortunately not, ive tried from the beginning much like yourself yet I'm still getting the same error, I think like someone said earlier it's to do with the raspberry Pi IP address. Going to have a tinker with it again this weekend!

**Comment by opes** [comment]
I had this same issue. It appears to be a regression from: https://github.com/rgraciano/echo-sonos/commit/5490c9f4fd8923eb6e8e1d296e5e09f3dceff0df

After undoing all the changes in that commit, SQS is successfully working again. Hope this helps, @jameshiscott & @youdy

**Comment by youdy** [comment]
@opes Just had a go of reverting the changes back to that commit and the error has disappeared however im now getting a different error, I think it may be to do with my settings. I will try again at the weekend from fresh and see if I can get it working again. Alexa does recognise the command now though!

**Comment by jameshiscott** [comment]
@opes how do I download the older codebase?

**Comment by pheintzelman** [comment]
Would be nice to fix this issue in the master branch. I took a quick look and the issue didn't jump out to me. Has anyone else taken a look at the code in the commit?

**Comment by jplourde5** [comment]
I'll take a look at it tomorrow to find out what it the newer echo-Sonos master broke the SQS and/or DynamoDB calls.

**Comment by jameshiscott** [comment]
Cool, thanks, happy to test for you if you find something tomorrow

**Comment by jplourde5** [comment]
Looks like some major rework has been performed with the newer version abstracting the AWS calls.  Looking at it now...

**Comment by jplourde5** [comment]
There is a typo in line 345 in sonosProxyFactory   useSqs should be useSQS

**Comment by jameshiscott** [comment]
Nice. Always the simple things....

**Comment by pheintzelman** [comment]
@jplourde5  :)

**Comment by rgraciano** [comment]
Merged fix in #164
